lib/ui/profitloss/RecordTransaction.dart

```diff
@@ -287,8 +347,8 @@ class RecordTransactionState extends State<RecordTransaction> {
         child: RecordTransactionListItem(
           viewModel: RecordTransactionListItemViewModel(
             type: RecordCellType.description,
-            isEditable: viewModel.isEditable,
-            description: viewModel.isEditable
+            isEditable: isEditing || isNewRecord,
+            description: isEditing
                 ? userData.description
                 : viewModel.actions[_Constants.thirdListItem].description,
           ),
```

Comment:
I know you make few changes here, but how you would see to change this huge method to something like this?

```
List<Widget> _buildContent(
      RecordTransactionViewModel viewModel, RecordTransactionStyle style) {
    List<Widget> listBuilder = [
      createRecordTransactionHeader(viewModel),
      SizedBox(height: style.headerLineSpace),
      createRecordTransactionListItemDatePicker(viewModel),
      ListDivider.build(),
      recordTransactionListItemPickItem(viewModel),
      ListDivider.build(),
      createRecordTransactionListItemDescription(viewModel)
    ];

    return listBuilder;
  }

  Padding createRecordTransactionListItemDescription(
      RecordTransactionViewModel viewModel) {
    return Padding(
      padding: _Constants.descriptionPadding,
      child: RecordTransactionListItem(
        viewModel: RecordTransactionListItemViewModel(
          type: RecordCellType.description,
          isEditable: isEditing || isNewRecord,
          description: isEditing
              ? userData.description
              : viewModel.actions[_Constants.thirdListItem].description,
        ),
        style: RecordTransactionListItemStyles.biggerStyle,
        parent: this,
      ),
    );
  }

  RecordTransactionListItem recordTransactionListItemPickItem(
      RecordTransactionViewModel viewModel) {
    return RecordTransactionListItem(
      viewModel: RecordTransactionListItemViewModel(
        type: RecordCellType.pickItem,
        isEditable: isEditing || isNewRecord,
        selectedItem: isEditing
            ? userData.crop
            : viewModel.actions[_Constants.secondListItem].selectedItem,
        listOfCrops: viewModel.actions[_Constants.secondListItem].listOfCrops,
      ),
      parent: this,
    );
  }

  RecordTransactionListItem createRecordTransactionListItemDatePicker(
      RecordTransactionViewModel viewModel) {
    return RecordTransactionListItem(
      viewModel: RecordTransactionListItemViewModel(
        type: RecordCellType.pickDate,
        isEditable: isEditing || isNewRecord,
        selectedDate: isEditing
            ? userData.date
            : viewModel.actions[_Constants.firstListItem].selectedDate,
      ),
      parent: this,
    );
  }

  RecordTransactionHeader createRecordTransactionHeader(
      RecordTransactionViewModel viewModel) {
    return RecordTransactionHeader(
        viewModel: RecordTransactionHeaderViewModel(
          isEditable: isEditing || isNewRecord,
          amount: viewModel.amount,
        ),
        style: viewModel.type == TransactionType.sale
            ? RecordTransactionHeaderStyles.defaultSaleStyle
            : RecordTransactionHeaderStyles.defaultCostStyle,
        parent: this);
  }
```

Reply:
this widget should be fully refactored